birdhouse/optional-components/README.rst

```diff
@@ -443,3 +443,38 @@ How to enable X-Robots-Tag Header in ``env.local`` (a copy from `env.local.examp
 
     .. seealso::
         See the `env.local.example`_ file for more details about this ``BIRDHOUSE_PROXY_ROOT_LOCATION`` behaviour.
+
+Prometheus Long-term Metrics
+----------------------------
+
+This is a second prometheus instance that collects longterm monitoring metrics from the original prometheus instance
+(the one created by the ``components/monitoring`` component).
+
+Longterm metrics are any prometheus rule that have the label ``group: longterm-metrics`` or in other words are
+selectable using prometheus' ``'{group="longterm-metrics"}'`` query filter. To see which longterm metric rules are
+added by default see the ``optional-components/prometheus-longterm-metrics/config/monitoring/prometheus.rules.template``.
+
+To configure this component:
+
+    * update the ``PROMETHEUS_LONGTERM_RETENTION_TIME`` variable to set how long the data will be kept by prometheus
+    * update the ``PROMETHEUS_LONGTERM_STORE_INTERVAL`` variable to set how often the longterm metrics rules will be
+      calculated. For example, setting it to ``10h`` will calculate these metrics every 10 hours.
+
+Enabling this component creates the additional endpoint ``/prometheus-longterm-metrics``.
+
+Thanos
+------
+
+This enables better storage of longterm metrics collected by the ``optional-components/prometheus-longterm-metrics``
+component. Data will be collected from the ``prometheus-longterm-metrics`` and stored in an S3 object store
+indefinitely.
```

Comment:
Oh I see, is there a switch to disable Thanos?

Same question, in the case we want to use Thanos, how to visualize the data stored on Thanos?  I assume if Thanos is enabled, the retention duration on the Prometheus side will be very short to avoid doubling the storage so without data being stored in Prometheus, how to visualize that data stored on Thanos.

Just a question.  If another component is required,  we can do it in a follow up Pr.

Reply:
> Oh I see, is there a switch to disable Thanos?

To answer my own question, Thanos is actually a separate component so it does not have to be enabled together with the Prometheus-long-term component?  The Prometheus-long-term component can function standalone of Thanos?

Reply:
> Thanos is actually a separate component so it does not have to be enabled together with the Prometheus-long-term component

Yes that's right. `prometheus-longterm-metrics` collects and stores specific metrics that we want to keep for longer from `prometheus`. If you want to also enable `thanos`, then `thanos` will store those same metrics in a much more compact/efficient way so that you can store more data over a longer time period.

Reply:
Let's see how much space that will take in practice and we will adjust.  And eventually we need a way to visualize those older metrics.  Otherwise, what's the point to keep them forever if we can not visualize?